[CI] MixedClusterClientYamlTestSuiteIT class failing

**Build Scans:**
- [elasticsearch-intake #11540 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/7jgkwznhnqpws)
- [elasticsearch-pull-request #37419 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/thob2mhat672u)
- [elasticsearch-intake #11533 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4jf3hpuyavr76)
- [elasticsearch-pull-request #37409 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/f722dg2rszg2u)
- [elasticsearch-pull-request #37408 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/3d6x3pmpjui32)
- [elasticsearch-intake #11532 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/snezjvl2ca5j4)
- [elasticsearch-pull-request #37098 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/vnfklfpq6tntw)
- [elasticsearch-pull-request #37042 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/q3qinvidymmjm)
- [elasticsearch-pull-request #36953 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/vgwcxjixg7pog)
- [elasticsearch-pull-request #36890 / 8.17.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/olvn75getvfg6)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.17.0#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=range/20_synthetic_source/Integer range}" -Dtests.seed=7FF66C42F583428F -Dtests.bwc=true -Dtests.locale=nds-Latn-DE -Dtests.timezone=Australia/North -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [range/20_synthetic_source:87]: Expected a map containing
integer_range: a map containing
            gte: expected <2> but was <missing>
            lte: <3>
             gt: <unexpected> but was <1>
```

**Issue Reasons:**
- [main] 3 consecutive failures in class org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
- [main] 3 consecutive failures in step 8.17.0_bwc-snapshots
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 3 consecutive failures in pipeline elasticsearch-pull-request
- [main] 13 failures in class org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT (2.3% fail rate in 565 executions)
- [main] 11 failures in step 8.17.0_bwc-snapshots (3.7% fail rate in 298 executions)
- [main] 2 failures in step 8.16.0_bwc-snapshots (1.1% fail rate in 185 executions)
- [main] 3 failures in pipeline elasticsearch-intake (2.1% fail rate in 146 executions)
- [main] 10 failures in pipeline elasticsearch-pull-request (2.9% fail rate in 347 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -361,8 +361,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/115129
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   issue: https://github.com/elastic/elasticsearch/issues/115135
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  issue: https://github.com/elastic/elasticsearch/issues/115213
 
 # Examples:
 #

--- a/qa/mixed-cluster/build.gradle
+++ b/qa/mixed-cluster/build.gradle
@@ -71,6 +71,7 @@ BuildParams.bwcVersions.withWireCompatible { bwcVersion, baseName ->
       numberOfNodes = 4
       setting 'path.repo', "${buildDir}/cluster/shared/repo/${baseName}"
       setting 'xpack.security.enabled', 'false'
+      setting "xpack.license.self_generated.type", "trial"
       /* There is a chance we have more master changes than "normal", so to avoid this test from failing,
        we increase the threshold (as this purpose of this test isn't to test that specific indicator). */
       if (bwcVersion.onOrAfter(Version.fromString("8.4.0"))) {

